### DISCUSSION TITLE
Support for expressions (JSONPath, JMESPath or similar) when scrubbing.

### DISCUSSION THREAD

**Discussion by bjorkstromm** [VerifyTests/Verify#1165] [Ideas]
Consider having a JSON like

```json
{
  "id": "foo",
  "c": {
    "id": "bar"
  }
}
```

Now if I want to scrub `id` in the inner object `c`, there's no easy way to do it using only JSON. The below will scrub all properties with name `id`.

```csharp
await VerifyJson(json).ScrubMember("id");
```

I propose adding support for JSONPath, JMESPath (or similar) expressions for more fine-grained control on which members to scrub. In the above example, if I only want the inner `id` scrubbed I could do something like:

```csharp
await VerifyJson(json).ScrubMemberMatching("c.id");
```

**Comment by bjorkstromm** [comment]
Forgot to mention, but if you'd consider this a useful feature @SimonCropp, I'd be more than happy to try submitting a PR.

**Reply by SimonCropp** [reply]
happy to consider a PR
